### DISCUSSION TITLE
Reduce serverless function size for dynamic route

### DISCUSSION THREAD

**Discussion by 7alip** [vercel/next.js#24774] [Help] [answered]
Hi folks!

I have been struggling with Serverless Function file size limitation. 

```
Error: Required Serverless Function files exceed max size of 50000000 bytes.
Initial Required compressed file size: 88974694
```

I've switched from `next-mdx-enhanced` to `next-mdx-remote`. Instead of using mdx files under page folder, I'm using `[...slug].tsx` **dynamic i18n featured 3-level nested routing**. I have more than 30 pages x 3 locales = 150 pages. Does `getStaticProps` create serverless functions seperately for each page?

**Is there any way to reduce the serverless function size?**

### Mdx Content Structure

```
- content
  - [course]
    - [module]
      - [slug]
        - [locale].mdx
```

### GetStaticPaths

```tsx
export const getStaticPaths: GetStaticPaths = async () => {
  const mdxPages = loadMDXInfo(CONTENT_PATH)

  const paths = mdxPages.map(({ slug }) => {
    const slugArr = slug.split('/').filter(Boolean)

    // /developer/module-1/first-topic/en.mdx
    const [course, module, url, lang] = slugArr

    return {
      params: {
        slug: [course, module, url],
      },
      locale: lang,
    }
  })

  return {
    paths,
    fallback: false,
  }
}
```

### GetStaticProps

```tsx
export const getStaticProps: GetStaticProps = async ({ params, locale }) => {
  const { slug } = params
  const filePath = typeof slug === 'string' ? slug : slug.join('/')

  const mdxFilePath = path.join(CONTENT_PATH, `${filePath}/${locale}.mdx`)

  const source = fs.readFileSync(mdxFilePath)

  const { content, data } = matter(source)

  const mdxSource = await serialize(content)

  const ssrTranslations = await serverSideTranslations(locale, ['common'])

  return {
    props: {
      source: mdxSource,
      frontMatter: data,
      course: slug[0],
      filePath: `/courses/${filePath}`,
      ...ssrTranslations,
    },
  }
}
```

**Comment by bobarros** [comment, marked as answer]
Do you have a serverless function of almost 90Mb? 

I might sound very stupid, because I don't have any knowledge about your project, but are you sure that everything it's working as it should? This is a huge serverless function.

Are you not loading videos, images, or something that you shouldn't in these props?

**Reply by 7alip** [reply]
I've noticed that unnecessary one of devDependencies caused that. It's solved once I removed. Thanks.
